lib/active_admin/resource.rb

```diff
@@ -135,11 +135,19 @@ def defined_actions
       controller.instance_methods.map(&:to_sym) & ResourceController::ACTIVE_ADMIN_ACTIONS
     end
 
-    def belongs_to(target, options = {})
-      @belongs_to = Resource::BelongsTo.new(self, target, options)
+    def belongs_to(targets, options = {})
+      @belongs_to = Resource::BelongsTo.new(self, targets, options.dup)
```

Comment:
Do the `options` get mutated now or why are we dupping them?

Reply:
Yes, it get mutated.

Reply:
mmm, but if it's the `controller` mutating them, we are already passing a mutated hash to it, so I guess this one shouldn't be needed.

Reply:
I remember the reason why, the controller mutates `class_name` option, which interferes with `BelongsTo`. For multiple optional resources, we don't want `class_name` to be set, because it will mess up the `targets` array.

Reply:
But the controller has its own copy of this hash, right? I just tried to run the specs without passing mutated hashes the `BelongsTo.new` and they all passed so if that's really need I think we need some extra coverage somewhere.

Reply:
Let me check if this is a problem without duplicating the hash.

Reply:
I haven't tested it yet, but I think that if you don't duplicate the `options` hash and have the code reloaded the second time, then the `BelongsTo#targets` will be incorrect.

Reply:
I'll like into that, thanks!